### PR TITLE
feat: Receive rsm 034 in json

### PR DESCRIPTION
**Pull request by x-platformcoder**
## Description

Makes it possible to request changes to charge notifications (D18 & D08) as json instead of xml

## References

* #1670

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargePrices.cs [hunk 1]
```diff
@@ -259,6 +260,35 @@ public async Task Given_ChargePriceMessageWithChargeInformationData_WhenPosted_T
                 notification.Should().NotContain("<cim:taxIndicator>");
             }
 
+            [Fact(Skip = "Used for debugging ChargeCommandReceivedEventHandler")]
```
Comment (Mech0z) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargePrices.cs:263]: Should we add tests only to skip them?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargePrices.cs [hunk 2]
```diff
@@ -481,6 +511,38 @@ public async Task When_TaxTariffPricesAreUpdatedBySystemOperator_Then_ANotificat
...
+
+                // Assert
+                actualResponse.StatusCode.Should().Be(HttpStatusCode.Accepted);
+
+                using var assertionScope = new AssertionScope();
+                var peekResult = await Fixture.MessageHubMock.AssertPeekReceivesRepliesAsync(correlationId, 8, ResponseFormat.Json);
+                peekResult.Should().NotContainMatch("*RejectRequestChangeOfPriceList_MarketDocument*");
+                peekResult.Should().ContainMatch("*NotifyPriceList_MarketDocument*");
+                peekResult.Should().ContainMatch("*\"value\": \"D08\"*");
+                peekResult.Should().NotContainMatch("*\"value\": \"D18\"*");
```
Comment (Mech0z) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargePrices.cs:532]: Should we group
Should Contain and Should not contain for easier readability

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Bundling/JsonBundleCreator.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+using System.IO;
+using System.Threading.Tasks;
+using Energinet.DataHub.MessageHub.Client.Storage;
+using Energinet.DataHub.MessageHub.Model.Model;
+using GreenEnergyHub.Charges.MessageHub.Infrastructure.Cim;
+using GreenEnergyHub.Charges.MessageHub.Models.AvailableData;
+
+namespace GreenEnergyHub.Charges.MessageHub.Infrastructure.Bundling
+{
+    public class JsonBundleCreator<TAvailableData> : IBundleCreator
```
Comment (Mech0z) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Bundling/JsonBundleCreator.cs:24]: Should we rename the existing BundleCreator to XMLBundleCreator?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Cim/Bundles/Charges/ChargeCimJsonSerializer.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+using System.Text.Json.Nodes;
+using GreenEnergyHub.Charges.Domain.Dtos.SharedDtos;
+using GreenEnergyHub.Charges.Infrastructure.Core.Cim.Charges;
+using GreenEnergyHub.Charges.Infrastructure.Core.Cim.MarketDocument;
+using GreenEnergyHub.Charges.MessageHub.Models.AvailableChargeData;
+using NodaTime;
+
+namespace GreenEnergyHub.Charges.MessageHub.Infrastructure.Cim.Bundles.Charges
+{
+    public class ChargeCimJsonSerializer : CimJsonSerializer<AvailableChargeData>
```
Comment (Mech0z) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Cim/Bundles/Charges/ChargeCimJsonSerializer.cs:25]: Same as above comment, should we rename the xml serializer

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/MessageHub/MessageHubHelper.cs [hunk 5]
```diff
@@ -27,13 +26,14 @@ public static class MessageHubHelper
         public static async Task<List<string>> AssertPeekReceivesRepliesAsync(
             this MessageHubMock messageHubMock,
             string correlationId,
-            int noOfDataAvailableNotifications = 1)
+            int noOfDataAvailableNotifications = 1,
+            ResponseFormat responseFormat = ResponseFormat.Xml)
```
Comment (Mech0z) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/MessageHub/MessageHubHelper.cs:30]: I would prefer if this was not a default parameter, to make it more explicit in tests if they are either json or xml

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargePrices.cs (hunk 2 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargePrices.cs:532]: done, made it consistent with other tests in the solution, where testing on cim elements in the log using contain and notcontain

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargePrices.cs (hunk 1 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargePrices.cs:263]: done, deleted (tested elsewhere)

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Bundling/JsonBundleCreator.cs (hunk 3 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Bundling/JsonBundleCreator.cs:24]: done

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Cim/Bundles/Charges/ChargeCimJsonSerializer.cs (hunk 4 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Cim/Bundles/Charges/ChargeCimJsonSerializer.cs:25]: done

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/MessageHub/MessageHubHelper.cs (hunk 5 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/MessageHub/MessageHubHelper.cs:30]: done

**Review by Mech0z — APPROVED** [review]

**Pull request merged**